src/check-dsl.ts

```diff
@@ -17,136 +20,106 @@ export const checkDSL = (codeInEditor: string) => {
   const reporter = report({ lines, markers });
 
   try {
-    const results = parseDSL(codeInEditor);
-    const relationsPerType: Record<string, string[]> = {};
-    const globalRelations = [Keywords.SELF as string];
+    const parserResults = parseDSL(codeInEditor);
+    const relationsPerType: Record<string, Record<string, boolean>> = {};
+    const globalRelations: Record<string, boolean> = { [Keywords.SELF]: true };
 
-    // Reading relations per type
-    _.forEach(results, (r) => {
-      const { typeName, typeDefinitionLine } = readTypeData(r, lines);
+    // Looking at the types
+    parserResults.forEach((typeDef) => {
+      const typeName = typeDef.type;
 
       // Include keyword
-      const relations = [Keywords.SELF as string];
+      const encounteredRelationsInType: Record<string, boolean> = { [Keywords.SELF]: true };
 
-      _.forEach(r[2], (r2, idx: number) => {
-        const relation = _.trim(_.flattenDeep(r2).join("")).match(/define\s+(.*)\s+as\s/)?.[1];
-        if (!relation) {
-          return;
-        }
-        const lineIdx = typeDefinitionLine + idx + 1;
+      typeDef.relations.forEach((relationDef) => {
+        const { relation: relationName } = relationDef;
 
-        if (relations.includes(relation)) {
-          reporter.duplicateDefinition({ lineIndex: lineIdx, value: relation });
+        // Check if we have any duplicate relations
+        if (encounteredRelationsInType[relationName]) {
```

Comment:
how does this work? `encounteredRelationsInType` was defined in line 32 but you never added any values to it?